public/script.js

```diff
@@ -1,23 +1,71 @@
-// Find the button
-// when you click the button, it should call a function "getSnippet"
+// all elements for the script from the index .
+const inputs = document.getElementById('paragraphInput');
 const paragraph = document.getElementById('text-snippet');
-
 const button = document.getElementById('generate-snippet-btn');
+
+// event listener for click events.
 button.addEventListener('click', getSnippet);
 
+// get snippet function 
 function getSnippet() {
-    // call the server using "fetch" to '/generateText' API route
-    // wait for it to return response (read about **promises** and **async await**), take the text that it provided
-    // find the "text-snippet" element and set it's text to the response's text
+    // getting the value of the input.
+    const amountOfSnippetsToGenerate = inputs.value
+    const requestPayload = {score: amountOfSnippetsToGenerate}
+
+    // speech settings
     let speech = new SpeechSynthesisUtterance();
     speech.lang = "en";
 
-    fetch('/generateText')
+    // detects number && letters
+    let letters = /^[A-Za-z]+$/;
+    let numbers = /^[0-9]+$/;
+    
+    // options for fetch
+    const options = {
+      method: "POST",
+      body: JSON.stringify(requestPayload) ,
+      headers: {"Content-type": "application/json; charset=UTF-8"}
+    }
+
+    // checks for letters and empty spaces, and bug preventing ..
+    if(amountOfSnippetsToGenerate === ""){
+        setErrorFor( inputs, 'please enter a number')
+    }
+    else if (amountOfSnippetsToGenerate.match(letters)){
+        setErrorFor( inputs, 'please enter a number')
+    }
+    else if (amountOfSnippetsToGenerate > 10 ){
+        setErrorFor( inputs, 'cannot ask more then 10')
+    }
+    else if (amountOfSnippetsToGenerate <= 0 ){
+        setErrorFor( inputs, 'cannot ask 0 generates')
+    }
```

Comment:
What if I will change the code in my browser and remove all these limitations? Then the server will crash if I send infinite number. You must do this validation in server side as well.